### PR TITLE
Adjust cmake to install same .h and .pc files as configure

### PR DESCRIPTION
**Pull request by dankegel**
This is the first few commits of https://github.com/zlib-ng/zlib-ng/pull/639

They're the easy part of the fix for #638 

The rest is touchier and deserves its own pull request (possibly after https://github.com/zlib-ng/zlib-ng/pull/560 lands)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -969,6 +975,28 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
...
+endif()
+if (HAVE_UNISTD_H)
+  SET(ZCONF_UNISTD_LINE "#if 1    /* was set to #if 1 by ./configure */")
+else()
+  SET(ZCONF_UNISTD_LINE "#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by ./configure */")
+endif()
+if (NEED_PTRDIFF_T)
+    SET(ZCONF_PTRDIFF_LINE "#if 1    /* was set to #if 1 by ./configure */")
+else()
+    SET(ZCONF_PTRDIFF_LINE "#ifdef NEED_PTRDIFF_T    /* may be set to #if 1 by ./configure */")
```
Comment (Dead2) [inline CMakeLists.txt]: I am thinking this maybe should not reference ./configure specifically..
Should we change that (also in configure) to `may be set to #if 1 during build`, or change `./configure` to `configure/cmake`?

**Comment by Dead2** [comment]
Other than the inline comment being misleading, the rest looks good to me.

File: .github/workflows/pkgcheck.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: CI pkgcheck
```
Comment (nmoinvaz) [inline .github/workflows/pkgcheck.yml]: Capitailize `PkgCheck` like the rest of the CI workflows..

File: .github/workflows/pkgcheck.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [
+          Ubuntu Gcc
+        ]
+        include:
+          - name: Ubuntu Gcc
```
Comment (nmoinvaz) [inline .github/workflows/pkgcheck.yml]: Capitialize GCC like the rest of the CI workflows..

File: CMakeLists.txt [hunk 4]
```diff
@@ -40,7 +40,7 @@ set(INSTALL_BIN_DIR "${CMAKE_INSTALL_PREFIX}/bin" CACHE PATH "Installation direc
 set(INSTALL_LIB_DIR "${CMAKE_INSTALL_PREFIX}/lib" CACHE PATH "Installation directory for libraries")
 set(INSTALL_INC_DIR "${CMAKE_INSTALL_PREFIX}/include" CACHE PATH "Installation directory for headers")
 set(INSTALL_MAN_DIR "${CMAKE_INSTALL_PREFIX}/share/man" CACHE PATH "Installation directory for manual pages")
-set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
+set(INSTALL_PKGCONFIG_DIR "${INSTALL_LIB_DIR}/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
```
Comment (nmoinvaz) [inline CMakeLists.txt:43]: Why this change? Minizip also seems to install into `/share/pkgconfig`..

File: CMakeLists.txt [hunk 5]
```diff
@@ -968,6 +974,28 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
     endif()
 endif()
 
+if (HAVE_STDARG_H)
```
Comment (nmoinvaz) [inline CMakeLists.txt]: Remove spaces after `if` keyword.

File: CMakeLists.txt [hunk 6]
```diff
@@ -968,6 +974,28 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
...
+  SET(ZCONF_STDARG_LINE "#if 1    /* was set to #if 1 by configure/cmake/etc */")
+else()
+  SET(ZCONF_STDARG_LINE "#ifdef HAVE_STDARG_H    /* may be set to #if 1 by configure/cmake/etc */")
+endif()
+if (HAVE_UNISTD_H)
```
Comment (nmoinvaz) [inline CMakeLists.txt]: Remove space after `if` keyword.

File: CMakeLists.txt [hunk 7]
```diff
@@ -968,6 +974,28 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
...
+  SET(ZCONF_UNISTD_LINE "#if 1    /* was set to #if 1 by configure/cmake/etc */")
+else()
+  SET(ZCONF_UNISTD_LINE "#ifdef HAVE_UNISTD_H    /* may be set to #if 1 by configure/cmake/etc */")
+endif()
+if (NEED_PTRDIFF_T)
```
Comment (nmoinvaz) [inline CMakeLists.txt]: Remove space after `if` keyword.

File: CMakeLists.txt [hunk 8]
```diff
@@ -968,6 +974,28 @@ if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
...
+    SET(ZCONF_PTRDIFF_LINE "#if 1    /* was set to #if 1 by configure/cmake/etc */")
+else()
+    SET(ZCONF_PTRDIFF_LINE "#ifdef NEED_PTRDIFF_T    /* may be set to #if 1 by configure/cmake/etc */")
+endif()
+
+set(ZLIB_PC ${CMAKE_CURRENT_BINARY_DIR}/zlib${SUFFIX}.pc)
```
Comment (nmoinvaz) [inline CMakeLists.txt:993]: Doesn't appear anything was changed in this block, maybe just moved down? Correct?

**Review by nmoinvaz — COMMENTED** [review]
Nice work, made a few suggestions..

File: CMakeLists.txt (hunk 4 above)
Comment (dankegel) [reply CMakeLists.txt:43]: Because configure installs .pc files into INSTALL_LIB_DIR, and cmake should match configure's behavior.  (Also, this matches current linux conventions as far as I know, both debian and red hat.)

File: CMakeLists.txt (hunk 8 above)
Comment (dankegel) [reply CMakeLists.txt:993]: Right, moved down until after all the variables used in zlib.pc.cmakein have been defined
(particularly ZLIB_FULL_VERSION about 50 lines up).

File: .github/workflows/pkgcheck.yml (hunk 2 above)
Comment (dankegel) [reply .github/workflows/pkgcheck.yml]: Done (though no intercaps, since the script pkgcheck.sh is all lower case, it'd be strange to capitalize the c)

File: .github/workflows/pkgcheck.yml (hunk 3 above)
Comment (dankegel) [reply .github/workflows/pkgcheck.yml]: Done.

File: CMakeLists.txt (hunk 5 above)
Comment (dankegel) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 6 above)
Comment (dankegel) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 7 above)
Comment (dankegel) [reply CMakeLists.txt]: Done.

Comment (dankegel) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 1 above)
Comment (dankegel) [reply CMakeLists.txt]: Done.

File: CMakeLists.txt (hunk 4 above)
Comment (nmoinvaz) [reply CMakeLists.txt:43]: Hmm ok. I may have to change minizip then. Good to know.

**Comment by dankegel** [comment]
The coverage failure seems spurious... some of the ppc jobs didn't run?

**Comment by dankegel** [comment]
I think this is ready to merge.

**Review by nmoinvaz — APPROVED** [review]

**Comment by Dead2** [comment]
> 
> 
> The coverage failure seems spurious... some of the ppc jobs didn't run?

The coverage comments are incomplete, you need to follow the #642 link in the comment, because on the website it is correct. This PR currently shows zero coverage change.
We have an ongoing ticket with codecov to get this fixed here: https://community.codecov.io/t/github-pr-not-being-updated-with-final-coverage-data/1425/25

Hopefully they will figure out what is wrong soon.

**Comment by Dead2** [comment]
Looks very good :)
